src/passes/Precompute.cpp

```diff
@@ -829,7 +829,7 @@ struct Precompute
       return true;
     }
     // We can emit a StringConst for a string constant.
-    if (type.isString()) {
+    if (type.isRef() && type.getHeapType().isMaybeShared(HeapType::string)) {
```

Comment:
This verbosity is a pretty big downside to being explicit, I think. How about making things more uniform by adding such shorthand for other types? I mean that alongside `type.isString()` we could have `type.isI31()` and so forth? I think that would make for a much nicer reading experience.

Reply:
The verbosity is unfortunate, but I think it's the right choice. With a method like `isString()`, it's unclear from just reading the use site whether it includes shared strings, whether it includes both nullable and non-nullable references, etc. If we add a new dimension like shareability in the future, we would want to update the shorthand methods to include it, but that update might not be correct for some users. When you have to spell it out explicitly, there's much less room for error.

Reply:
I agree the verbosity is less risky, but I also think there is a natural interpretation here: `isString/isArray/etc.` should refer to either shared or non-shared. That is just a consistent extension of how they handle another property of types, nullability: `isString/isArray/etc.` ignore nullability, and likewise it is natural for them to ignore shareability.

In other words, I don't think this has been confusing at all regarding nullability, so I am not worried about shareability.